### PR TITLE
Fixes and tweaks

### PR DESCRIPTION
**Pull request by hharen**
Tweaks and fixes in this pull request: 
- more appropriate font sizes 
- presentation slides increased size
- add a pointer to the lesson topic on the left panel
- add the opening of external links in the new window (done with javascript, not ideal but trix doesn't give an option to add target to the link) 
- add `first?` and `last?` methods to show _previous/next_ links on lesson view
- add validation of uniqueness of user on subscription
- fix showing of the 'You have no courses' notice on Subscription index view

### CODE REVIEW

File: app/models/lesson.rb [hunk 1]
```diff
@@ -15,24 +15,41 @@ class Lesson < ApplicationRecord
...
   private
 
-  def find_neighbouring_lesson(direction)
+  def neighbouring_lesson(direction)
     lessons_ids = course.topics.map(&:lessons).flatten.pluck(:id)
     lesson_index = lessons_ids.index(id)
+    new_lesson_index = lesson_index + direction
+
+    return nil if new_lesson_index.negative?
+    return nil if (new_lesson_index) > (lessons_ids.count - 1)
```
Comment (nerdinand) [inline app/models/lesson.rb:45]: Are the parentheses necessary here? I'd leave them off...

File: app/models/lesson.rb [hunk 2]
```diff
@@ -15,24 +15,41 @@ class Lesson < ApplicationRecord
...
+
+    get_lesson(lessons_ids, new_lesson_index)
+  end
+
+  def get_lesson(lessons_ids, new_lesson_index)
     Lesson.find(lessons_ids[
-      (lesson_index + direction).clamp(0, lessons_ids.length - 1)
+      new_lesson_index.clamp(0, lessons_ids.length - 1)
```
Comment (nerdinand) [inline app/models/lesson.rb:52]: Technically, now the `clamp`ing isn't necessary anymore. You do that manually on lines 44 and 45. And then there's not much point in this method. You can just inline it.

File: app/models/subscription.rb [hunk 3]
```diff
@@ -5,6 +5,8 @@ class Subscription < ApplicationRecord
   belongs_to :course
   has_many :lessons, through: :course
 
+  validates :user, uniqueness: true
```
Comment (nerdinand) [inline app/models/subscription.rb:8]: Doesn't this mean that each user can just have a single Subscription? I think only the combination of user and course should be unique.

File: app/models/lesson.rb (hunk 2 above)
Comment (hharen) [reply app/models/lesson.rb:52]: That's true but I'll miss the `clamp` ;)

File: app/models/subscription.rb (hunk 3 above)
Comment (hharen) [reply app/models/subscription.rb:8]: 😱 you're totally right, thanks 😄

**Pull request merged**